From d5482aa07972efe63b38898eaf68e6f7036ef48e Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Fri, 12 Feb 2021 08:31:57 -0600
Subject: [PATCH 4/4] Revert "radeon, amdgpu: Firmware is required for DRM and
 KMS on R600 onward"

Although this patch helps in Debian, not all distributions using Zen
Kernrel have the same folder structure (/lib/firmware).  And as @Atemu
pointed out in #200, the patch is technically incredibly old for kernel
standards so it may only do harm than good.  Revert this patch.

This reverts commit 789af38db7fe47a34799bb71a0dfde6cb38767b7.
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 29 ------------------------
 drivers/gpu/drm/radeon/radeon_drv.c     | 30 -------------------------
 2 files changed, 59 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index ed03b2712..0b786d8dd 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -36,8 +36,6 @@
 #include <linux/vga_switcheroo.h>
 #include <drm/drm_probe_helper.h>
 #include <linux/mmu_notifier.h>
-#include <linux/namei.h>
-#include <linux/path.h>
 
 #include "amdgpu.h"
 #include "amdgpu_irq.h"
@@ -1100,28 +1098,6 @@ MODULE_DEVICE_TABLE(pci, pciidlist);
 
 static struct drm_driver kms_driver;
 
-/* Test that /lib/firmware/amdgpu is a directory (or symlink to a
- * directory).  We could try to match the udev search path, but let's
- * keep it simple.
- */
-static bool amdgpu_firmware_installed(void)
-{
-#if IS_BUILTIN(CONFIG_DRM_AMDGPU)
-	/* It may be too early to tell.  Assume it's there. */
-	return true;
-#else
-	struct path path;
-
-	if (kern_path("/lib/firmware/amdgpu", LOOKUP_DIRECTORY | LOOKUP_FOLLOW,
-		      &path) == 0) {
-		path_put(&path);
-		return true;
-	}
-
-	return false;
-#endif
-}
-
 static int amdgpu_pci_probe(struct pci_dev *pdev,
 			    const struct pci_device_id *ent)
 {
@@ -1186,11 +1162,6 @@ static int amdgpu_pci_probe(struct pci_dev *pdev,
 	}
 #endif
 
-	if (!amdgpu_firmware_installed()) {
-		DRM_ERROR("amdgpu requires firmware installed\n");
-		return -ENODEV;
-	}
-
 	/* Get rid of things like offb */
 	ret = drm_fb_helper_remove_conflicting_pci_framebuffers(pdev, "amdgpudrmfb");
 	if (ret)
diff --git a/drivers/gpu/drm/radeon/radeon_drv.c b/drivers/gpu/drm/radeon/radeon_drv.c
index 8add4be70..4cd30613f 100644
--- a/drivers/gpu/drm/radeon/radeon_drv.c
+++ b/drivers/gpu/drm/radeon/radeon_drv.c
@@ -49,8 +49,6 @@
 #include <drm/drm_probe_helper.h>
 #include <drm/drm_vblank.h>
 #include <drm/radeon_drm.h>
-#include <linux/namei.h>
-#include <linux/path.h>
 
 #include "radeon_drv.h"
 
@@ -316,28 +314,6 @@ static struct drm_driver kms_driver;
 
 bool radeon_device_is_virtual(void);
 
-/* Test that /lib/firmware/radeon is a directory (or symlink to a
- * directory).  We could try to match the udev search path, but let's
- * keep it simple.
- */
-static bool radeon_firmware_installed(void)
-{
-#if IS_BUILTIN(CONFIG_DRM_RADEON)
-	/* It may be too early to tell.  Assume it's there. */
-	return true;
-#else
-	struct path path;
-
-	if (kern_path("/lib/firmware/radeon", LOOKUP_DIRECTORY | LOOKUP_FOLLOW,
-		      &path) == 0) {
-		path_put(&path);
-		return true;
-	}
-
-	return false;
-#endif
-}
-
 static int radeon_pci_probe(struct pci_dev *pdev,
 			    const struct pci_device_id *ent)
 {
@@ -378,12 +354,6 @@ static int radeon_pci_probe(struct pci_dev *pdev,
 	if (vga_switcheroo_client_probe_defer(pdev))
 		return -EPROBE_DEFER;
 
-	if ((ent->driver_data & RADEON_FAMILY_MASK) >= CHIP_R600 &&
-	    !radeon_firmware_installed()) {
-		DRM_ERROR("radeon kernel modesetting for R600 or later requires firmware installed\n");
-		return -ENODEV;
-	}
-
 	/* Get rid of things like offb */
 	ret = drm_fb_helper_remove_conflicting_pci_framebuffers(pdev, "radeondrmfb");
 	if (ret)
-- 
2.30.0.452.gfb7fa4a1fd

